File: Strategy/Java/src/MiniDuckSimulator.java
package eu.svetlin.design.patterns.strategy;

import eu.svetlin.design.patterns.strategy.duck.Duck;
import eu.svetlin.design.patterns.strategy.duck.MallardDuck;
import eu.svetlin.design.patterns.strategy.duck.ModelDuck;
import eu.svetlin.design.patterns.strategy.duck.RubberDuck;
import eu.svetlin.design.patterns.strategy.fly.FlyRocketPowered;

public class MiniDuckSimulator {
    public static void main(String[] args){
        System.out.println("Hello duck!\n");
        Duck mallard = new MallardDuck();
        mallard.display();
        mallard.performQuack();
        mallard.performFly();

        System.out.println();

        Duck rubber = new RubberDuck();
        rubber.display();
        rubber.performQuack();
        rubber.performFly();

        System.out.println();

        Duck model = new ModelDuck();
        model.display();
        model.performFly();
        model.setFlyBehavior(new FlyRocketPowered());
        model.performFly();
    }
}

File: Strategy/TypeScript/src/miniDuckSimulator.ts
import MallardDuck from './duck/MallardDuck';
import ModelDuck from './duck/ModelDuck';
import RubberDuck from './duck/RubberDuck';
import FlyRocketPowered from './fly/FlyRocketPowered';

const main = () => {
  console.log('Hello duck\n');
  const mallard = new MallardDuck();
  mallard.display();
  mallard.performQuack();
  mallard.performFly();

  console.log('\n');

  const rubber = new RubberDuck();
  rubber.display();
  rubber.performQuack();
  rubber.performFly();

  console.log('\n');

  const model = new ModelDuck();
  model.display();
  model.performQuack();
  model.performFly();
  model.setFlyBehavior(new FlyRocketPowered());
  model.performFly();
};

main();
